addons/mail/static/src/core/messaging_service.js

```diff
@@ -205,6 +205,34 @@ export class Messaging {
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this.threadService.insert({
+                            id: notif.payload.id,
+                            model: "discuss.channel",
+                        });
+                        this.store.discuss.starred.messages =
+                            this.store.discuss.starred.messages.filter(
+                                (msg) => msg.resModel !== thread.model || msg.resId !== thread.id
+                            );
```

Comment:
These filter would probably be easier to read with using `removeFromArrayWithPredicate`.

Reply:
IMO using `removeFromArrayWithPredicate` will be more costlier. Because finding each element through the array traversing it every time and then removing it individually would not be as efficient as just replacing the array with new one.

Reply:
I don't think there is a performance problem here, but it's fine for me either way, it's a small detail.